test/gas/EndToEnd.t.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {MockERC20} from "../../lib/solady/test/utils/mocks/MockERC20.sol";
+import {CoinbaseSmartWallet} from "../../src/CoinbaseSmartWallet.sol";
+import {CoinbaseSmartWalletFactory} from "../../src/CoinbaseSmartWalletFactory.sol";
+import {SmartWalletTestBase} from "../CoinbaseSmartWallet/SmartWalletTestBase.sol";
+
+import {Static} from "../CoinbaseSmartWallet/Static.sol";
+import {MockTarget} from "../mocks/MockTarget.sol";
+import {UserOperation} from "account-abstraction/interfaces/UserOperation.sol";
+import {console2} from "forge-std/Test.sol";
+
+/// forge-config: default.isolate = true
+contract EndToEndTest is SmartWalletTestBase {
+    address eoaUser = address(0xe0a);
+
+    MockERC20 usdc;
+    MockTarget target;
+    CoinbaseSmartWalletFactory factory;
+
+    function setUp() public override {
+        vm.etch(0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789, Static.ENTRY_POINT_BYTES);
+
+        CoinbaseSmartWallet implementation = new CoinbaseSmartWallet();
+        factory = new CoinbaseSmartWalletFactory(address(implementation));
+
+        signerPrivateKey = 0xa11ce;
+        signer = vm.addr(signerPrivateKey);
+        owners.push(abi.encode(signer));
+        account = factory.createAccount(owners, 0);
+
+        vm.deal(address(account), 100 ether);
+        vm.deal(eoaUser, 100 ether);
+
+        usdc = new MockERC20("USD Coin", "USDC", 6);
+        usdc.mint(address(account), 10000e6);
+        usdc.mint(eoaUser, 10000e6);
+
+        target = new MockTarget();
+    }
+
+    // Native ETH transfer comparison between ERC-4337 and EOA
+    function test_transfer_native() public {
```

Comment:
would benefit from comments to annotate the code sections here for easier skimming